### ISSUE TITLE
Unexpected behaviors in WHERE [PATH_PATTERN]

### ISSUE THREAD

**Issue by yaochen625** [neo4j/neo4j#13235]
- Neo4j version: 5.8.0 enterprise Docker
- Operating system: Ubuntu22.04
- API/Driver: Cypher Driver

# Introduction

All the following queries should be executed successfully, but they have different behavior on empty and nonempty databases. I test it on empty database and system database.

```cypher
MATCH (n) WHERE ( :PERSON)-[ :KNOWS]-( :PERSON) RETURN *
```

```cypher
MATCH (n) WHERE ( :PERSON) RETURN *
```

```cypher
MATCH (n) WHERE () RETURN *
```

```cypher
MATCH (n) WHERE (n) RETURN *
```

## Analysis # 1

Let's take a look at the following query

```cypher
MATCH (n) WHERE ( :PERSON)-[ :KNOWS]-( :PERSON) RETURN *
```

In an empty database, its result is empty, which is satisfactory.

<img width="884" alt="image" src="https://github.com/neo4j/neo4j/assets/71594689/26fb4ebc-0fe0-413c-93ce-3d92c3025c3b">

But in a non-empty database, it will trigger an exception. And the exception is interesting, saying that it is executing the procedure call, which is apparently wrong.
<img width="893" alt="image" src="https://github.com/neo4j/neo4j/assets/71594689/d124c384-6a5b-4404-b814-5c36208fcd9f">

## Analysis # 2 

Let's take a look at the following query

```cypher
MATCH (n) WHERE ( :PERSON) RETURN *
```

In empty databases, it triggered an exception, however, the similar query in Anaysis#1 did not trigger.

<img width="892" alt="image" src="https://github.com/neo4j/neo4j/assets/71594689/6dde843b-640f-44e4-8314-a46d1b5a718b">

In non-empty database, it also triggered an exception, but it is different from all the exception mentioned above

<img width="871" alt="image" src="https://github.com/neo4j/neo4j/assets/71594689/cfaad409-9809-4a40-8393-0b0bbf9e96f7">

## Analysis # 3 

Let's take a look at the following query

```cypher
MATCH (n) WHERE () RETURN *
```

In empty databases, it triggered an exception, which leads to confusion. In the specification, it should mean a empty pattern.

<img width="891" alt="image" src="https://github.com/neo4j/neo4j/assets/71594689/397d8659-551c-4732-b22a-95006e292e1f">

In the non-empty database, it also triggered an exception.

<img width="862" alt="image" src="https://github.com/neo4j/neo4j/assets/71594689/51b54083-9988-48f1-aacc-6f3afb7d4374">

## Analysis # 4

Let's take a look at the following query

```cypher
MATCH (n) WHERE (n) RETURN *
```

n is also part of a pattern, it should return the correct answer but it regards it as Node

<img width="873" alt="image" src="https://github.com/neo4j/neo4j/assets/71594689/922bde65-9717-4756-85e9-3da33a9b1ce6">

**Comment by pontusmelke** [comment]
Hi @HITFAY, thanks for reporting.

- Analysis 1. The second query is run on the system database which is not allowed. Please do `:use neo4j` and run the query again.
- Analysis 2. This is not a valid query, you can do `WHERE n:PERSON` if you want to check if the node `n` has the label `PERSON` 
- Analysis 3. A single node pattern is not supported as a predicate in a WHERE clause, furthermore having an empty node pattern is also unclear what that would mean in this query?
- Analysis 4. As mentioned above a single node pattern is not supported as a predicate in a where clause. If it would have been supported it would always evaluate to true here so it would not add much value. It also makes queries hard to read and hard to parse since you are allowed to add a parenthesis around boolean predicates, `WHERE a` is equivalent to  `WHERE (a)`. Should `(a)` here be interpreted as a single boolean value or as a node pattern?

Hope that helps!

Best regards,
Pontus

**Comment by joyemang33** [comment]
Hi! @pontusmelke 
I am also interested in these single-node patterns in the WHERE clauses.
I found that in the Neo4j docs https://neo4j.com/docs/cypher-manual/current/clauses/where/, there is no description of that single-node patterns in the WHERE clauses is disallowed. It may cause confusion to users, and these behaviors are unexpected in the current stage.

Would you want to modify the documents to make it more clear? 
However, I think it is better to support and parse the single node pattern in the WHERE clauses in the future release, by simply checking ```(a)``` is a node variable or a single bool value. It can also allow users to write ```(n : [label-expression])``` as a formal format in the WHERE clauses.

Best regards,
Joye

**Comment by pontusmelke** [comment]
Hi @joyemang33,
yes we will look into if we can make documentation more clear around this.

Regards,
Pontus

**Comment by joyemang33** [comment]
> Hi @joyemang33, yes we will look into if we can make documentation more clear around this.
> 
> Regards, Pontus

Thanks! Looking forward to seeing more explanations about that in the document.

**Comment by hvub** [comment]
You can have single node pattern in WHERE with the help of EXISTS subqueries. They won't be made available directly, because of the involved syntax ambiguity to existing expressions.
